src/crypto/poly1305.cpp

```diff
@@ -0,0 +1,198 @@
+/*
+ * Public Domain poly1305 from Andrew Moon
+ * poly1305-donna-unrolled.c from https://github.com/floodyberry/poly1305-donna
+ */
+
+/* $OpenBSD: poly1305.c,v 1.3 2013/12/19 22:57:13 djm Exp $ */
+
+#include <crypto/poly1305.h>
+
+#define mul32x32_64(a, b) ((uint64_t)(a) * (b))
+
+#define U8TO32_LE(p)                                                           \
+  (((uint32_t)((p)[0])) | ((uint32_t)((p)[1]) << 8) |                          \
+   ((uint32_t)((p)[2]) << 16) | ((uint32_t)((p)[3]) << 24))
+
+#define U32TO8_LE(p, v)                                                        \
+  do {                                                                         \
+    (p)[0] = (uint8_t)((v));                                                   \
+    (p)[1] = (uint8_t)((v) >> 8);                                              \
+    (p)[2] = (uint8_t)((v) >> 16);                                             \
+    (p)[3] = (uint8_t)((v) >> 24);                                             \
+  } while (0)
+
+void poly1305_auth(unsigned char out[POLY1305_TAGLEN], const unsigned char *m,
```

Comment:
Why did you remove the [poly1305_init](https://github.com/floodyberry/poly1305-donna/blob/master/poly1305-donna-32.h), [poly1305_update](https://github.com/floodyberry/poly1305-donna/blob/master/poly1305-donna.c#L27), [poly1305_finish](https://github.com/floodyberry/poly1305-donna/blob/master/poly1305-donna-32.h#L134) separation?

Reply:
Which separation? I took Andrew Moons implementation found at https://github.com/openssh/openssh-portable/blob/90e51d672711c19a36573be1785caf35019ae7a8/poly1305.c

Reply:
Ah, I see, in that case the source code URL at the top of the file is confusing.